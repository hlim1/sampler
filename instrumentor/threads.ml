open Cil


let threads =
  Options.registerBoolean
    ~flag:"threads"
    ~desc:"create thread-safe code"
    ~ident:"Threads"
    ~default:false


let bump file =
  if !threads then
    let bumper = Lval (var (FindFunction.find "atomicIncrementCounter" file)) in
    fun counter location ->
      Call (None, bumper, [mkAddrOf counter], location)
  else
    fun counter location ->
      Set (counter, increm (Lval counter) 1, location)
