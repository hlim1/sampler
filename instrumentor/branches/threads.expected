threads	branches	threads.c	11	drop_x	6	tmp	1	17000
threads	branches	threads.c	14	drop_x	14	x % 10 == 0	15300	1700
threads	branches	threads.c	22	drop_y	6	tmp	1	12000
threads	branches	threads.c	25	drop_y	14	y % 5 == 0	9600	2400
